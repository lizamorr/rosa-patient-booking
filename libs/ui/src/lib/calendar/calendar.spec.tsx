import { render } from '@testing-library/react';

import Calendar from './calendar';

describe('Calendar', () => {
  const calendarProps = {
    availabilities: [
      {
        _id: '62347ff3b433476d7d16bac1',
        eventIds: [],
        motiveIds: [
          '61379ba159d4940022b6c929',
          '61eea350ddf6c500149ae2cb',
          '61eea367ddf6c500149ae2cc',
        ],
        calendarId: '61379ba159d4940022b6c928',
        nextAvailabilityId: '',
        previousAvailabilityId: '',
        appointmentSlotId: '62347ff3b433476d7d16babb',
        meridiem: 'am',
        state: 'open',
        duration: 125,
        endAt: '2022-03-24T15:30:00.000Z',
        startAt: '2022-03-24T13:25:00.000Z',
        createdAt: '2022-03-18T12:49:55.665Z',
        updatedAt: '2022-03-18T12:49:55.665Z',
        day: 24,
        month: 2,
        year: 2022,
        dayOfTheWeek: 4,
        __v: 0,
        id: '62347ff3b433476d7d16bac1',
      },
      {
        _id: '62399aa1a845d7aa3f965106',
        eventIds: [],
        motiveIds: [
          '61379ba159d4940022b6c929',
          '61eea350ddf6c500149ae2cb',
          '61eea367ddf6c500149ae2cc',
        ],
        calendarId: '61379ba159d4940022b6c928',
        previousAvailabilityId: '62348018b433476d7d16bb0d',
        appointmentSlotId: '62348018b433476d7d16bb07',
        meridiem: 'am',
        state: 'open',
        duration: 135,
        endAt: '2022-03-25T10:30:00.000Z',
        startAt: '2022-03-25T08:15:00.000Z',
        dayOfTheWeek: 5,
        year: 2022,
        month: 2,
        day: 25,
        createdAt: '2022-03-22T09:45:05.999Z',
        updatedAt: '2022-03-22T09:45:05.999Z',
        __v: 0,
        id: '62399aa1a845d7aa3f965106',
      },
      {
        _id: '623a647bc47ecca3b8926222',
        eventIds: [],
        motiveIds: [
          '61379ba159d4940022b6c929',
          '61eea350ddf6c500149ae2cb',
          '61eea367ddf6c500149ae2cc',
        ],
        calendarId: '61379ba159d4940022b6c928',
        nextAvailabilityId: '62348031b433476d7d16bb26',
        appointmentSlotId: '62348031b433476d7d16bb20',
        meridiem: 'am',
        state: 'open',
        duration: 150,
        endAt: '2022-03-29T11:30:00.000Z',
        startAt: '2022-03-29T09:00:00.000Z',
        dayOfTheWeek: 2,
        year: 2022,
        month: 2,
        day: 29,
        createdAt: '2022-03-23T00:06:19.630Z',
        updatedAt: '2022-03-23T00:06:19.630Z',
        __v: 0,
        id: '623a647bc47ecca3b8926222',
      },
      {
        _id: '623b4434c47ecca3b892b1dd',
        eventIds: [],
        motiveIds: [
          '61379ba159d4940022b6c929',
          '61eea350ddf6c500149ae2cb',
          '61eea367ddf6c500149ae2cc',
        ],
        calendarId: '61379ba159d4940022b6c928',
        nextAvailabilityId: '623a647bc47ecca3b8926223',
        previousAvailabilityId: '62348031b433476d7d16bb26',
        appointmentSlotId: '62348031b433476d7d16bb20',
        meridiem: 'am',
        state: 'open',
        duration: 90,
        endAt: '2022-03-29T15:00:00.000Z',
        startAt: '2022-03-29T13:30:00.000Z',
        dayOfTheWeek: 2,
        year: 2022,
        month: 2,
        day: 29,
        createdAt: '2022-03-23T16:00:52.631Z',
        updatedAt: '2022-03-23T16:00:52.631Z',
        __v: 0,
        id: '623b4434c47ecca3b892b1dd',
      },
    ],
    meetingDuration: 75,
    selectedOption: '61eea350ddf6c500149ae2cb',
    appointmentDate: {
      day: 'Fri',
      meetings: [
        {
          day: 'Fri',
          time: '9:15',
          selected: true,
        },
      ],
    },
    hasRequiredError: false,
    onSetStartDate: jest.fn(),
    onSetEndDate: jest.fn(),
    onSetHasRequiredError: jest.fn(),
    onSetAppointmentDate: jest.fn(),
  };
  it('should match snapshot', () => {
    const component = render(<Calendar {...calendarProps} />);
    expect(component).toMatchSnapshot();
  });
});
