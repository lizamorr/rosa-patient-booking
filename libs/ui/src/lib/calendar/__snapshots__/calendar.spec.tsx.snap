// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calendar"
      >
        <table
          cellpadding="0"
          cellspacing="0"
        >
          <thead>
            <tr>
              <th
                class="arrow-icon__container"
              >
                <i
                  class="arrow-icon fa-solid fa-chevron-left disabled"
                />
              </th>
              <th>
                <div
                  class="calendar__header-container"
                >
                  <span
                    class="calendar__header-day"
                  >
                    Th
                  </span>
                  Mar
                   
                  
                  24
                </div>
              </th>
              <th>
                <div
                  class="calendar__header-container"
                >
                  <span
                    class="calendar__header-day"
                  >
                    Fr
                  </span>
                  Mar
                   
                  
                  25
                </div>
              </th>
              <th>
                <div
                  class="calendar__header-container"
                >
                  <span
                    class="calendar__header-day"
                  >
                    Sa
                  </span>
                  Mar
                   
                  
                  26
                </div>
              </th>
              <th>
                <div
                  class="calendar__header-container"
                >
                  <span
                    class="calendar__header-day"
                  >
                    Su
                  </span>
                  Mar
                   
                  
                  27
                </div>
              </th>
              <th>
                <div
                  class="calendar__header-container"
                >
                  <span
                    class="calendar__header-day"
                  >
                    Mo
                  </span>
                  Mar
                   
                  
                  28
                </div>
              </th>
              <th>
                <div
                  class="calendar__header-container"
                >
                  <span
                    class="calendar__header-day"
                  >
                    Tu
                  </span>
                  Mar
                   
                  
                  29
                </div>
              </th>
              <th
                class="arrow-icon__container"
              >
                <i
                  class="arrow-icon fa-solid fa-chevron-right"
                />
              </th>
            </tr>
          </thead>
          <tbody
            class=""
          >
            <tr>
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
            </tr>
            <tr>
              <td />
              <td />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calendar"
    >
      <table
        cellpadding="0"
        cellspacing="0"
      >
        <thead>
          <tr>
            <th
              class="arrow-icon__container"
            >
              <i
                class="arrow-icon fa-solid fa-chevron-left disabled"
              />
            </th>
            <th>
              <div
                class="calendar__header-container"
              >
                <span
                  class="calendar__header-day"
                >
                  Th
                </span>
                Mar
                 
                
                24
              </div>
            </th>
            <th>
              <div
                class="calendar__header-container"
              >
                <span
                  class="calendar__header-day"
                >
                  Fr
                </span>
                Mar
                 
                
                25
              </div>
            </th>
            <th>
              <div
                class="calendar__header-container"
              >
                <span
                  class="calendar__header-day"
                >
                  Sa
                </span>
                Mar
                 
                
                26
              </div>
            </th>
            <th>
              <div
                class="calendar__header-container"
              >
                <span
                  class="calendar__header-day"
                >
                  Su
                </span>
                Mar
                 
                
                27
              </div>
            </th>
            <th>
              <div
                class="calendar__header-container"
              >
                <span
                  class="calendar__header-day"
                >
                  Mo
                </span>
                Mar
                 
                
                28
              </div>
            </th>
            <th>
              <div
                class="calendar__header-container"
              >
                <span
                  class="calendar__header-day"
                >
                  Tu
                </span>
                Mar
                 
                
                29
              </div>
            </th>
            <th
              class="arrow-icon__container"
            >
              <i
                class="arrow-icon fa-solid fa-chevron-right"
              />
            </th>
          </tr>
        </thead>
        <tbody
          class=""
        >
          <tr>
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
          </tr>
          <tr>
            <td />
            <td />
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
